modules/hal_silabs/gecko/CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+# Makefile - Gecko SDK
+#
+# Copyright (c) 2017, Christian Taedcke
+# Copyright (c) 2021, Safran Passenger Innovations Germany GmbH
+# Copyright (c) 2022, Antmicro <www.antmicro.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(EMLIB_DIR ${ZEPHYR_HAL_SILABS_MODULE_DIR}/gecko/emlib)
+set(COMMON_DIR ${ZEPHYR_HAL_SILABS_MODULE_DIR}/gecko/common)
+set(DEVICE_DIR ${ZEPHYR_HAL_SILABS_MODULE_DIR}/gecko/Device)
+set(RADIO_DIR ${ZEPHYR_HAL_SILABS_MODULE_DIR}/gecko/platform/radio)
+set(BLOBS_DIR ${ZEPHYR_HAL_SILABS_MODULE_DIR/zephyr/blobs})
+
+# Translate the SoC name and part number into the gecko device and cpu name
+# respectively.
+string(TOUPPER ${CONFIG_SOC_SERIES} SILABS_GECKO_DEVICE)
+
+# Get SoC series number, i.e. translate e.g. efr32bg22 -> 22
+string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 GECKO_SERIES_NUMBER)
+
+set(SILABS_GECKO_PART_NUMBER ${CONFIG_SOC_PART_NUMBER})
+
+function(add_prebuilt_library lib_name prebuilt_path)
+  add_library(${lib_name} STATIC IMPORTED GLOBAL)
+  set_target_properties(${lib_name} PROPERTIES
+    IMPORTED_LOCATION ${BLOBS_DIR}/${prebuilt_path}
+  )
+  zephyr_link_libraries(${lib_name})
+endfunction()
+
+if(${CONFIG_SOC_GECKO_HAS_RADIO})
+  if(${CONFIG_SOC_FAMILY_SILABS_S1})
+    zephyr_include_directories(
+      ${RADIO_DIR}/rail_lib/plugin/pa-conversions/efr32xg1x/config
+      ${RADIO_DIR}/rail_lib/chip/efr32/efr32xg1x
+    )
+  endif()
+
+  zephyr_include_directories(
+    ${RADIO_DIR}/rail_lib/common
+    ${RADIO_DIR}/rail_lib/plugin/pa-conversions
+  )
+
+  if(CONFIG_SOC_GECKO_USE_RAIL)
+    # rail
+    zephyr_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_conversions_efr32.c)
+    zephyr_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_curves_efr32.c)
+
+    # prebuilt libs
+    add_prebuilt_library(librail platform/radio/rail_lib/autogen/librail_release/librail_efr32xg${GECKO_SERIES_NUMBER}_gcc_release.a)
```

Comment:
binary blob alert?

Reply:
> binary blob alert?

That file is pulled in by `west blobs fetch hal_silabs`.